src/components/swap/SwapBuyFiatButton.tsx

```diff
@@ -130,7 +137,12 @@ export default function SwapBuyFiatButton() {
         element={InterfaceElementName.FIAT_ON_RAMP_BUY_BUTTON}
         properties={{ account_connected: !!account }}
       >
-        <StyledTextButton onClick={handleBuyCrypto} disabled={buyCryptoButtonDisabled} data-testid="buy-fiat-button">
+        <StyledTextButton
+          onClick={handleBuyCrypto}
+          disabled={buyCryptoButtonDisabled}
+          data-testid="buy-fiat-button"
+          shouldFadeIn={shouldFadeIn.current}
```

Comment:
👀 why is this using a ref instead of state?

Reply:
using a ref means it will retain the value from initial render, which is the intention. It should not use state, as it won't be updated after it is initially set. See https://github.com/Uniswap/interface/pull/7393#discussion_r1342779363.